### PR TITLE
Place ships

### PR DESCRIPTION
**Pull request by Safisyx**
It took me much more time that I was expecting, but finally we can place ships or remove ships and send the board through the button when we are done placing all ships. :punch: 
We still need to implement the action of sending the board to the backend though :jack_o_lantern:
Also the CSS is still not pretty, but we can do that after the demo :stuck_out_tongue: 

:exclamation: I couldn't make small commits as I was working with all files at the same time so sorry! :smile:
Please ask me if there is anything unclear. 

The state for placing ships:
state-for-placing-ships

### CODE REVIEW

**Review by Safisyx — COMMENTED** [review]
For sure the code is a big mess but I just wanted to get the things running :stuck_out_tongue:

File: src/actions/placeShipInit.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {PLACE_SHIP_INIT} from './types'
+
```
Comment (Ju1111) [inline src/actions/placeShipInit.js:2]: isn't this the same action as PLACE_SHIP?

File: src/actions/sendBoard.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {SEND_BOARD} from './types'
+
```
Comment (Ju1111) [inline src/actions/sendBoard.js:2]: I guess this is the one we still need to figure out XD 
I'd say it needs a patch or so. Update board? Or maybe create a new board? Looking forward to discussing that soon hehe :)

File: src/components/ships/Board.css [hunk 3]
```diff
@@ -1,5 +1,10 @@
 .Board {
   display: flex;
+  flex-direction: column;
```
Comment (Ju1111) [inline src/components/ships/Board.css:3]: ohhhh what does this do?

File: src/components/ships/Board.js [hunk 4]
```diff
@@ -20,25 +20,27 @@ export class Board extends PureComponent {
...
 
   renderSquare = rowIndex => (value, index) => {
+    //console.log(value);
     return (
       <Square
         key={ index }
-        value={ value }
-        x={ index }
-        y={ rowIndex }
+        value={`${value}`}
```
Comment (Ju1111) [inline src/components/ships/Board.js]: why does it need `${}` ? Maybe I should know this but yeah apparently I don't. 🐱

File: src/components/ships/Board.js [hunk 5]
```diff
@@ -20,25 +20,27 @@ export class Board extends PureComponent {
...
       <div className="Board">
-        { this.props.board1.map(this.renderRow) }
+        { this.props.board.map(this.renderRow) }
       </div>
     )
   }
 }
 
-const mapStateToProps = ({ board: { board1 }  }) => ({ board1 })
+const mapStateToProps = ({ shipBoard}) => ({ board: shipBoard })
```
Comment (Ju1111) [inline src/components/ships/Board.js:44]: This works like that? You don't call shipBoard in the class if I see it correctly?
Also shouldn't they be the other way around? 🤔

File: src/components/ships/LockButton.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
 import React, { PureComponent } from 'react'
-import PropTypes from 'prop-types'
```
Comment (Ju1111) [inline src/components/ships/LockButton.js:2]: why do you remove this?

File: src/components/ships/LockButton.js [hunk 7]
```diff
@@ -11,7 +12,10 @@ class LockButton extends PureComponent {
   // }
```
Comment (Ju1111) [inline src/components/ships/LockButton.js:12]: think we should have this here yeah

File: src/components/ships/Square.css [hunk 8]
```diff
@@ -13,3 +13,27 @@
   cursor: pointer;
   background-color: #004d4d;
 }
+
+.vh {
+  border-color: yellow;
```
Comment (Ju1111) [inline src/components/ships/Square.css:18]: maybe change this to 
border: 2px solid yellow
then it's thicker ;) and then people can see it. We can also make it green to show it's allowed

**Review by Ju1111 — COMMENTED** [review]
I know I have some comments but all in all woooooow you are awesome!

File: src/actions/placeShipInit.js (hunk 1 above)
Comment (Safisyx) [reply src/actions/placeShipInit.js:2]: Well, we need two different types in the reducers. So basically they are not the same :stuck_out_tongue:

File: src/actions/sendBoard.js (hunk 2 above)
Comment (Safisyx) [reply src/actions/sendBoard.js:2]: Yes indeed we still need to implement that! :smile:

File: src/components/ships/Board.css (hunk 3 above)
Comment (Safisyx) [reply src/components/ships/Board.css:3]: Like that we don't have to reverse x and y, x will always be row and y column

File: src/components/ships/Board.js (hunk 4 above)
Comment (Safisyx) [reply src/components/ships/Board.js]: Yes we can remove it, I was thinking that {value} was the problem before realizing that I need to transpose the rendering of the board

File: src/components/ships/Board.js (hunk 5 above)
Comment (Safisyx) [reply src/components/ships/Board.js:44]: In fact I use board in the class, we can have clean code indeed if we use shipBoard as well but we can work on that later :smile:

File: src/components/ships/LockButton.js (hunk 6 above)
Comment (Safisyx) [reply src/components/ships/LockButton.js:2]: Because we don't use PropType in the class

File: src/components/ships/LockButton.js (hunk 7 above)
Comment (Safisyx) [reply src/components/ships/LockButton.js:12]: Yeah we can add that later

File: src/components/ships/Square.css (hunk 8 above)
Comment (Safisyx) [reply src/components/ships/Square.css:18]: Or maybe red, put that in the todo list

File: src/actions/sendBoard.js (hunk 2 above)
Comment (Ju1111) [reply src/actions/sendBoard.js:2]: noted :)

File: src/components/ships/Board.css (hunk 3 above)
Comment (Ju1111) [reply src/components/ships/Board.css:3]: love it! will keep this in mind for future projects

File: src/components/ships/Board.js (hunk 5 above)
Comment (Ju1111) [reply src/components/ships/Board.js:44]: haha we should have a function to create a list with things to do later in GitHub 😆

File: src/components/ships/LockButton.js (hunk 6 above)
Comment (Ju1111) [reply src/components/ships/LockButton.js:2]: I thought we did haha oopsie

File: src/components/ships/LockButton.js (hunk 7 above)
Comment (Ju1111) [reply src/components/ships/LockButton.js:12]: echo >> .dolater 😆

File: src/components/ships/Square.css (hunk 8 above)
Comment (Ju1111) [reply src/components/ships/Square.css:18]: hahaha again XD but I'd say red signalises that it's forbidden. so mayyyybe not red.

File: src/components/ships/LockButton.js (hunk 7 above)
Comment (Safisyx) [reply src/components/ships/LockButton.js:12]: :joy:

**Pull request merged**